docs: Add detailed "What is conda-forge?" page (#2823)
conda-forge/conda-forge.github.io: docs/user/what-is-conda-forge.md

In other words, conda-forge is much more than `python` and `conda`: it distributes the entire dependency chain (including the compilers!) for pretty much anyone to be able to run their projects.

## Encompassing ecosystems across languages

Projects use different languages, and language ecosystems do not exist in isolation: Python, C++, R, Fortran, and potentially assets (like model weights) live all together in end-users applications, and conda-forge distributes all of them so that they can be used consistently altogether in end-users projects.

:::info Benefits
conda-forge provides a plethora of packages from different ecosystems, making sure they can be used all together.
:::

## Built on an open package format, with community-managed artifacts

The conda package manager and format were created to address packaging challenges for the NumPy stack and other software with native extensions—projects that mix Python with C, Fortran, Cython, C++, and interfaces to external libraries, and that are harder to ship with tools aimed at pure-Python packages alone.

> At the first PyData meetup, several developers asked Guido van Rossum how to fix Python packaging for the NumPy stack; his answer was to "solve the problem ourselves." Engineers at Continuum (now Anaconda, Inc.) took that seriously: they studied dpkg, rpm, pip/virtualenv, Homebrew, Nix, and other systems, drew on experience with Enthought Python Distribution (EPD), and created the conda package manager and conda environments. A fuller account of this history is in progress in the [conda-forge history article](https://github.com/conda-forge/conda-forge.github.io/pull/2298) (see also [this talk](https://www.youtube.com/watch?v=U2oa_RLbTVA) and [this talk](https://www.youtube.com/watch?v=gFEE3w7F0ww)).

Reviewer: This really needs to refer back to Travis' blogpost, and I don't like that it's reusing his language almost verbatim without marking it as a quote.

> While Anaconda's distribution is linked to specific legal Terms of Service, conda-forge reuses this package format in a community-run distribution where artifacts are published under BSD-3-Clause, alongside the original projects' licenses.

Reviewer: This formulation will need some more care, and likely also context, explanation & reshuffling. I imagine @jezdez will have opinions.

Just off-the-cuff, I'd probably start with something like
```suggestion
Condensing a lot of history, it's important to distinguish different meanings of the word "conda":
* the packaging paradigm that grew out of the above-mentioned efforts
* `conda`, the CLI tool used to interact with packages in this format
* Anaconda, the company where this was initially developed, which maintains curated channels of packages that may be subject to licensing fees (subject to the size of the organisation that users belong to)
* the conda ecosystem, which has outgrown the control of Anaconda, and nowadays is a [standards-based](https://github.com/conda/ceps/blob/main/cep-1.md) [effort](https://conda.org/blog/2024-08-14-conda-ecosystem-explained), under multi-stakeholder [governance](https://github.com/conda/governance).

To relate conda-forge to all this: it is the the community-driven channel of packages which are _not_ subject to Anaconda's Terms of Services, all the while still being hosted on Anaconda infrastructure (i.e. https://anaconda.org/conda-forge), which can be installed through `conda` as well as other installers like `mamba` & `pixi`, and which forms a key part of the wider conda ecosystem, in addition to being (broadly, but not fully) the "upstream" of Anaconda's own curated channels (roughly comparable for example to the dynamic between CentOS Stream and Redhat Enterprise Linux).
```

:::info Benefits
conda-forge's users aren't technically constrained by the operations of a single entity, and package usage is governed by artifact and upstream project licenses.
:::

## Open, automated, distributed and transparent maintenance of projects' distribution

> Anaconda distribution was really popular, but it was also faced with a tremendous number of requests to add or modify packages, so people decided to create a distributed alternative that could scale: conda-forge.

Reviewer: Same here; the formulation needs to be fair from all possible angles, including Anaconda's.
```suggestion
Under conda's packaging paradigm, libraries need to be "packaged" according to a declarative recipe that describes how the sources are processed, which tools and other libraries need to be present before running the build scripts, etc. These recipes form the backbone of the conda ecosystem.

The flipside however, is that in order to benefit from the advantages of conda's paradigm, such a recipe needs to be created initially and then maintained as new versions of the library get released. 

Given the success of conda's approach, this meant that users very soon wanted to consume a vast variety of packages through conda, and given the scope (not just Python, but many more language) of the task, as well as the very high rate of change, no single entity -- not even Anaconda -- could keep up with packaging everything users wanted, and keeping it up to date.

What ended up happening is that the volunteer-driven conda-forge channel could absorb contributions much more quickly and easily than an enterprise distribution, and eventually ended up becoming the place where thousands of contributors of varying degrees of involvement and expertise come together in a way that's proven most scalable in dealing with the vast breadth and depth and pace of the wider software ecosystem.
```

---
title: "What is conda-forge?"
---

# Why conda-forge as a software distribution?

> Conda-forge's mission is to make software easily distributable and usable by the world.

Reviewer: mission statements probably need a vote in core? 🤷 

I also would aim higher than "distributable".

Maybe:
```suggestion
Conda-forge's goal is to make software easily installable and usable by the world.
```

> conda-forge solves the problem of distributing scientific software properly and consistently for many applications and platforms.

Reviewer: We need to decide on a consistent capitalization pattern; above you started the sentence with "Conda-forge [...]"; this sentence is also a bit disjointed from the ones before & after.

```suggestion
Conda-forge grew out of the intersection between Python and the scientific software ecosystem, not least because it solves many problems in this space that had not been solved elsewhere. Scientific computing is still a core constituency of conda-forge, but over the last 10+ years, the ecosystem has grown to encompass packages from essentially all programming languages, across all major platforms and architectures.
```
We could also reference https://pypackaging-native.github.io/#key-issues here.

> conda-forge is much more than `python` and `conda`, and it covers much more than scientific software applications: conda-forge distributes the entire dependency chain for anyone to be able to run their projects.

Reviewer: ```suggestion
In other words, conda-forge is much more than `python` and `conda`: it distributes the entire dependency chain (including the compilers!) for pretty much anyone to be able to run their projects.
```

## Encompassing ecosystems across languages

Projects use different languages, and language ecosystems do not exist in isolation: Python, C++, R, Fortran, and potentially assets (like model weights) live all together in end-users applications.

> conda-forge distributes all of them so that they can be used consistently altogether in end-users projects.

Reviewer: remove the linebreak to the previous sentence here.

:::info Benefits
conda-forge provides a plethora of packages from different ecosystems, making sure they can be used all together.
:::

## Built on an open package format, with community-managed artifacts

> conda packages originated from maintainers of Scientific Python projects at Anaconda who wanted to properly solve distributing their projects given their complexity (mixture of Python, C, Fortran, Cython, C++ code; interface with external C and Fortran libraries) compared to normal python-only projects.

Reviewer: I'm not that deep into the history, but this sounds like an over-simplification; see https://github.com/conda-forge/conda-forge.github.io/pull/2298 (and [this](https://www.youtube.com/watch?v=U2oa_RLbTVA) or [this](https://www.youtube.com/watch?v=gFEE3w7F0ww) for yet more details); not that I expect you to absorb all that, but since this is going to be like the big neon sign explaining conda-forge, we need to get this right.

> Anaconda came up with a package format which could meet the needs of distributing Python projects with native extensions: the conda package.

Reviewer: I would also use a few more words here to describe this; e.g. something in the spirit of the following quote (from [here](https://technicaldiscovery.blogspot.com/2013/12/why-i-promote-conda.html)):

> This was confirmed for me at the first PyData meetup at Google HQ, where several of us asked Guido what we can do to fix Python packaging for the NumPy stack. Guido's answer was to "solve the problem ourselves".  We at Continuum took him at his word.  We looked at dpkg, rpm, pip/virtualenv, brew, nixos, and 0installer, and used our past experience with EPD.  We thought hard about the fundamental issues, and created the conda package manager and conda environments.

> While Anaconda's distribution is linked to specific legal Terms of Services, conda-forge reuse this package format but without any specific legal terms, but the original projects' license.

Reviewer 2: We do attach BSD-3-Clause on all the artifacts

Author: Rephrased with 35ea285.
